release/downloadIstioCandidate.sh

```diff
@@ -36,10 +36,10 @@ fi
 
 # Determine the latest Istio version by version number ignoring alpha, beta, and rc versions.
 if [ "x${ISTIO_VERSION}" = "x" ] ; then
-  ISTIO_VERSION="$(curl -sL https://github.com/istio/istio/releases | \
-                  grep -o 'releases/[0-9]*.[0-9]*.[0-9]*/' | sort --version-sort | \
-                  tail -1 | awk -F'/' '{ print $2}')"
-  ISTIO_VERSION="${ISTIO_VERSION##*/}"
+  response="$(curl -sL https://github.com/istio/istio/releases.atom)"
```

Comment:
Unfortunately, the title of the script does not match the desired function. I had tried to fix this in the past, but the PRs where never merged.  istio.io/downloadIstio points to this script and we have install instructions that say `curl -L https://istio.io/downloadIstio | sh -` so we only want releases version, non-rcs and non-betas for this command.

Reply:
See https://github.com/istio/istio/pull/17187 and https://github.com/istio/istio/pull/17178 for past discussion.

We should at least update the text at the top of this file (similar to what was in the above) to at least get the remarks correct as well.